### PR TITLE
Section container

### LINKED ISSUE #1675: Create ws frontpage section container
Blocked on https://github.com/bbc/simorgh/pull/1640 https://github.com/bbc/psammead/pull/486 

**Describe the solution you'd like**
Using the story promo container, story list component and section divider component (but not blocked on the divider) create a container to render a 'section'. This section will contain the data of a single cps group.

It will take in an array of items, in a group object, as seen below.

```
{
   "type": "top-stories",
   "title": "Top stories",
   "items": [...],
   "strapline": {
      "name": "nammmmeee",
   };
},
```

It will loop over the array of stories, creating StoryPromo container elements and inserting them into the story promo list.

It will also render a section divider at the top of the component (with a strapline title if there is one)

Note, this wont perfectly match the UX, e.g. strapline at the top of all sections, including the first 'top stories' one, however this will be interated on later.

Fixture can be imported direction from the `data` folder

**A11y:**
Does the section itself need any specific a11y logic? Are we using a `section` tag etc?

**Testing notes**
[Tester to complete]

Dev insight: Storybook and unit snapshots, maybe some assertion tests for good luck

**Additional context**

### PR DESCRIPTION
**Pull request by andrew-nowak**
Resolves #1675

**Overall change:** Create the first implementation of the Section Container

**Code changes:**

- Contains a `psammead-section-label`, and a `psammead-story-promo-list` of `psammead-story-promo`s
- Contained within a `<section>`, with the `aria-labelledby` attribute, pointing at the `psammead-section-label`
- If there is no strapline in the data, that section will not be rendered

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

**Comment by andrew-nowak** [comment]
~~blocked waiting for a non-alpha version of `@bbc/psammead-story-promo-list`~~

File: src/app/containers/FrontPageSection/index.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  const { script } = ctx;
+  const sectionLabelId = idSanitiser(group.title);
+
+  // The current implementation of SectionLabel *requires* a strapline to be
+  // present in order to render. It is currently *not possible* to render a
+  // section that does not have a strapline without breaking both the visual
+  // *and especially* the screen reader UX.
+  // If this group does not have a strapline; do not render!
+  // This may change in the future, if a way to avoid breaking UX is found.
+  if (!(group.strapline && group.strapline.name)) {
```
Comment (dr3) [inline src/app/containers/FrontPageSection/index.jsx]: Can you use `deepGet` here please

File: src/app/containers/FrontPageSection/index.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import { bool, shape } from 'prop-types';
+import SectionLabel from '@bbc/psammead-section-label';
+import { StoryPromoUl, StoryPromoLi } from '@bbc/psammead-story-promo-list';
+import { ServiceContext } from '../../contexts/ServiceContext';
+import StoryPromo from '../StoryPromo';
+import groupShape from '../../models/propTypes/frontPageGroup';
+import idSanitiser from '../../lib/utilities/idSanitiser';
+
+const FrontPageSection = ({ bar, group }) => {
+  const ctx = useContext(ServiceContext);
```
Comment (dr3) [inline src/app/containers/FrontPageSection/index.jsx]: You can combine these 2 lines with 
```
const { script } = useContext(ServiceContext);
```

**Review by dr3 — CHANGES_REQUESTED** [review]
Looks great, just a couple things

File: src/app/containers/FrontPageSection/index.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  const { script } = useContext(ServiceContext);
+  const sectionLabelId = idSanitiser(group.title);
+
+  // The current implementation of SectionLabel *requires* a strapline to be
+  // present in order to render. It is currently *not possible* to render a
+  // section that does not have a strapline without breaking both the visual
+  // *and especially* the screen reader UX.
+  // If this group does not have a strapline; do not render!
+  // This may change in the future, if a way to avoid breaking UX is found.
+  if (!deepGet(['strapline', 'name'], group)) {
```
Comment (dr3) [inline src/app/containers/FrontPageSection/index.jsx]: Ah soz, can you set this to a variable and use it on line 33 please?

File: src/app/containers/FrontPageSection/index.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    // (<section> tags *should* imply `role="region"`)
+    // While this may be true in a perfect world, we set it in order to get
+    // the greatest possible support.
+    // eslint-disable-next-line jsx-a11y/no-redundant-roles
+    <section role="region" aria-labelledby={sectionLabelId}>
+      <SectionLabel script={script} labelId={sectionLabelId} bar={bar}>
+        {group.strapline.name}
+      </SectionLabel>
+      <StoryPromoUl>
+        {group.items.map(item => (
```
Comment (dr3) [inline src/app/containers/FrontPageSection/index.jsx]: I dont think this checks that the key `items` exists. Can you do
```
const items = deepGet(['items'], group);
```
At the top and then throw null if it is falsey (inc empty)

**Review by dr3 — CHANGES_REQUESTED** [review]

**Review by dr3 — APPROVED** [review]
Great work, really nice comments and testing :100:

**Review by ghost — APPROVED** [review]
Great work.

**Review by DenisHdz — APPROVED** [review]
Awesome, we are getting there 🕺🏻

**Comment by jamesbrumpton** [comment]
LGTM. Happy for this to be merged.

**Pull request merged**